elks/fs/super.c

```diff
@@ -486,14 +486,22 @@ void mount_root(void)
     struct file *filp;
     int retval;
 
+    /* TODO: allow mounting root from both character and block devices */
+    /* TEMP: hack to allow mounting from /dev/rom */
+#ifdef CONFIG_ROMFS_FS
+    d_inode = new_inode(NULL, S_IFCHR);
```

Comment:
Shouldn't ROMFS work as a block device? If this is merged and ROMFS is enabled, doesn't it break non-ROMFS roots?

Reply:
No, ROMFS does not work neither with a block device nor with buffers, but accesses directly to the data in ROM. The root device `/dev/rom` is a dummy one, just because ELKS requires one to mount a filesystem.

But I agree the change in `mount_root()` is not clean and would break other use cases. Furthermore, this design does not comply to the Linux standard: https://www.kernel.org/doc/html/v4.11/admin-guide/devices.html

I then propose to remove that hack in `super.c`, to change `/dev/rom` from character to block device, and to allocate the (major,minor) = (31, 0). Do you agree ?

Reply:
It makes sense to me to make `/dev/rom` a real block device that consists of the contents of the computer's boot ROM and make ROMFS work on that `/dev/rom` block device without any special behavior. That would allow ROMFS to be used on other devices outside of `/dev/rom` if desired for some reason.

Reply:
OK, then, let me change `/dev/rom` to a block device. Working on it, will be part of the next commit.